apps/rush-lib/src/utilities/Utilities.ts

```diff
@@ -548,23 +622,42 @@ export class Utilities {
         continue;
       }
 
-      environment[key] = initialEnvironment[key];
+      environment[key] = options.initialEnvironment[key];
     }
 
-     // When NPM invokes a lifecycle script, it sets an environment variable INIT_CWD that remembers
-     // the directory that NPM started in.  This allows naive scripts to change their current working directory
-     // and invoke NPM operations, while still be able to find a local .npmrc file.  Although Rush recommends
-     // for toolchain scripts to be professionally written (versus brittle stuff like
-     // "cd ./lib && npm run tsc && cd .."), we support INIT_CWD for compatibility.
-     //
-     // More about this feature: https://github.com/npm/npm/pull/12356
-     if (initCwd) {
-       environment['INIT_CWD'] = initCwd; // tslint:disable-line:no-string-literal
-     }
+    // When NPM invokes a lifecycle script, it sets an environment variable INIT_CWD that remembers
+    // the directory that NPM started in.  This allows naive scripts to change their current working directory
+    // and invoke NPM operations, while still be able to find a local .npmrc file.  Although Rush recommends
+    // for toolchain scripts to be professionally written (versus brittle stuff like
+    // "cd ./lib && npm run tsc && cd .."), we support INIT_CWD for compatibility.
+    //
+    // More about this feature: https://github.com/npm/npm/pull/12356
+    if (options.initCwd) {
+      environment['INIT_CWD'] = options.initCwd; // tslint:disable-line:no-string-literal
+    }
+
+    if (options.pathOptions) {
+      if (options.pathOptions.includeRepoBin && options.pathOptions.commonTempFolder) {
+        environment.PATH = Utilities.addNodeModulesBinToPATH(environment.PATH, options.pathOptions.commonTempFolder);
+      }
+
+      if (options.pathOptions.includeProjectBin && options.pathOptions.projectRoot) {
+        environment.PATH = Utilities.addNodeModulesBinToPATH(environment.PATH, options.pathOptions.projectRoot);
+      }
+    }
 
     return environment;
   }
 
+  private static addNodeModulesBinToPATH(existingPATH: string | undefined, rootDirectory: string): string {
```

Comment:
According to our capitalization conventions the name should be `addNodeModulesBinToPath`